p2p/service.py

```diff
@@ -94,8 +97,21 @@ def __init__(self, token: CancelToken=None) -> None:
             if finished_callback is not None:
                 finished_callback(self)
 
+    def schedule_child_service(self, child_service: 'BaseService') -> None:
+        """
+        Schedule a child service to run with the main service. On shutdown the main service
```

Comment:
@gsalgado @pipermerriam let's do some bike shedding over the name.

The reason why I'd like to have `child` in their is because it implies that hierachy that we are essentially creating (child services run when main service runs, main service waits for child services to cleanup).

The reason why I use `schedule` rather than `run` is because this method really doesn't run the services yet but just puts them in a list. They are all run after the `_run()` method of the main service. 

Ok, but now that I'm thinking about that...*maybe* we don't want to run them immediately because if we run it directly we could also start such child services after `_run()` was called. That's not possible with the current design

How about I change that and then we call the method `run_child_service`?